mslib/mswms/dataaccess.py

```diff
@@ -322,23 +355,46 @@ def _add_to_filetree(self, filename, content):
     def setup(self):
         # Get a list of the available data files.
         self._available_files = [
-            _filename for _filename in sorted(os.listdir(self._root_path)) if self._domain_id in _filename]
+            _filename for _filename in os.listdir(self._root_path) if self._domain_id in _filename]
         logging.info("Files identified for domain '%s': %s",
                      self._domain_id, self._available_files)
 
+        for filename in list(self._file_cache):
+            if filename not in self._available_files:
+                del self._file_cache[filename]
+
         self._filetree = {}
         self._elevations = {"sfc": {"filename": None, "levels": [], "units": None}}
 
         # Build the tree structure.
         for filename in self._available_files:
-            logging.info("Opening candidate '%s'", filename)
-            try:
-                content = self._parse_file(filename)
-            except IOError as ex:
-                logging.error("Skipping file '%s' (%s: %s)", filename, type(ex), ex)
-                continue
-            if content["vert_type"] not in self._elevations:
-                self._elevations[content["vert_type"]] = content["elevations"]
+            mtime = os.path.getmtime(os.path.join(self._root_path, filename))
```

Comment:
I think we should try: .. getmtime except IOError. logging ...

Reply:
Maybe. In case that someone removed the file in between listing the directory and parsing them here, it will not work anyway...